bayesflow/datasets/offline_dataset.py

```diff
@@ -11,12 +12,20 @@ class OfflineDataset(keras.utils.PyDataset):
     See the `DiskDataset` class for handling large datasets that are split into multiple smaller files.
     """
 
-    def __init__(self, data: dict[str, np.ndarray], batch_size: int, adapter: Adapter | None, **kwargs):
+    def __init__(
+        self, data: dict[str, np.ndarray], batch_size: int, adapter: Adapter | None, num_samples: int = None, **kwargs
+    ):
         super().__init__(**kwargs)
         self.batch_size = batch_size
         self.data = data
         self.adapter = adapter
-        self.num_samples = next(iter(data.values())).shape[0]
+
+        if num_samples is None:
```

Comment:
I do like that the user has both options (a) to automatically compute num_samples (default) or (b) to manually specify it, if for some reason, num_samples cannot be reliably inferred by the data.